system/lvol.py

```diff
@@ -201,8 +213,22 @@ def main():
 
     lvs = parse_lvs(current_lvs)
 
+    if thinpool and lv:
+        for test_lv in lvs:
+            if test_lv['name'] == thinpool:
+                break
+        else:
+            module.fail_json(msg="Thin poll %s does not exist." % thinpool)
```

Comment:
I think the error message is wrong, shouldn't it be "Thin pool" rather than "Thin poll" ?